pkg/initializer/validator/datavalidator_test.go

```diff
@@ -41,170 +42,244 @@ var _ = Describe("Running Datavalidator", func() {
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(int(dataDirStatus)).Should(Equal(FailBelowRevisionConsistencyError))
 			})
-			Context("with consistent revision numbers between etcd and latest snapshot", func() {
-				Context("with corrupt data directory", func() {
-					Context("with corrupt db file", func() {
-						It("should return DataDirStatus as DataDirectoryCorrupt or DataDirectoryError or RevisionConsistencyError, and nil error", func() {
-							dbFile := path.Join(restoreDataDir, "member", "snap", "db")
-							_, err = os.Stat(dbFile)
-							Expect(err).ShouldNot(HaveOccurred())
-
-							tempFile := path.Join(outputDir, "temp", "db")
-							err = copyFile(dbFile, tempFile)
-							Expect(err).ShouldNot(HaveOccurred())
-
-							file, err := os.OpenFile(
-								dbFile,
-								os.O_WRONLY|os.O_TRUNC|os.O_CREATE,
-								0666,
-							)
-							Expect(err).ShouldNot(HaveOccurred())
-							defer file.Close()
-
-							// corrupt the db file by writing random data to it
-							byteSlice := []byte("Random data!\n")
-							_, err = file.Write(byteSlice)
-							Expect(err).ShouldNot(HaveOccurred())
-
-							dataDirStatus, err := validator.Validate(Full, 0)
-							Expect(err).ShouldNot(HaveOccurred())
-							Expect(int(dataDirStatus)).Should(SatisfyAny(Equal(DataDirectoryCorrupt), Equal(DataDirectoryError), Equal(RevisionConsistencyError)))
-
-							err = os.Remove(dbFile)
-							Expect(err).ShouldNot(HaveOccurred())
-
-							err = os.Rename(tempFile, dbFile)
-							Expect(err).ShouldNot(HaveOccurred())
-						})
-					})
-				})
-				Context("with clean data directory", func() {
-					Context("with fail below revision configured to low value", func() {
-						It("should return DataDirStatus as DataDirectoryValid, and nil error", func() {
-							dataDirStatus, err := validator.Validate(Full, 0)
-							Expect(err).ShouldNot(HaveOccurred())
-							Expect(int(dataDirStatus)).Should(Equal(DataDirectoryValid))
-						})
-					})
-
-					Context("with fail below revision configured to high value", func() {
-						It("should return DataDirStatus as FailBelowRevisionConsistencyError and nil error", func() {
-							validator.Config.SnapstoreConfig.Container = path.Join(snapstoreBackupDir, "tmp")
-							dataDirStatus, err := validator.Validate(Full, 1000000)
-							Expect(err).ShouldNot(HaveOccurred())
-							Expect(int(dataDirStatus)).Should(Equal(FailBelowRevisionConsistencyError))
-						})
-					})
-				})
+		})
+
+		Context("without providing snapstore config", func() {
+			It("should return DataDirStatus as DataDirectoryValid and nil error", func() {
+				validator.Config.SnapstoreConfig = nil
+				dataDirStatus, err := validator.Validate(Sanity, failBelowRevision)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(int(dataDirStatus)).Should(Equal(DataDirectoryValid))
 			})
 		})
 	})
+
+	Context("with failure on snapstore call", func() {
```

Comment:
```suggestion
	Context("with failure on snapstore call due to unknown snapstore provider", func() {
```
So that it'll be easy to differentiate between this and the next test context.